src/guides/v2.3/extension-dev-guide/routing.md

```diff
@@ -350,6 +350,75 @@ As a result, by accessing the `http://site.com/learning` route, the `http://site
 
 ![Routing Result]({{ site.baseurl }}/common/images/routing-result.png)
 
+## Declaring the new custom no route processor
+
+If the standard 404 page should be extended or the customer should be directed to another page, for example, the search page where the query path will be query content, the Magento allows adding a custom no route processor.
```

Comment:
Thank you for the clarification now it's much more helpful. Let's simplify this explanation a bit.

Thank you.

```suggestion
If the standard 404 page should be extended or the customer should be directed to another page, for example, the search page, Magento allows adding a custom no route processor.
```